File: src/main/java/com/indiepost/service/TagService.java
package com.indiepost.service;

import com.indiepost.dto.TagDto;

import java.util.List;

/**
 * Created by jake on 9/17/16.
 */
public interface TagService {
    TagDto save(TagDto tagDto);

    TagDto findById(Long id);

    TagDto findByName(String name);

    List<String> findAllToStringList();

    List<TagDto> findAll(int page, int maxResults);

    List<TagDto> findByIds(List<Long> ids);

    void update(TagDto tagDto);

    void deleteById(Long id);

    List<TagDto> findByNameIn(List<String> tagNames);

    List<TagDto> findByNameLike(String tagName);
}

File: src/main/kotlin/com/indiepost/service/TagService.kt
package com.indiepost.service

import com.indiepost.dto.SelectedTagDto
import com.indiepost.dto.TagDto
import com.indiepost.enums.Types
import com.indiepost.model.Tag
import org.springframework.data.domain.Pageable

/**
 * Created by jake on 9/17/16.
 */
interface TagService {
    fun save(tag: Tag)

    fun findById(id: Long): Tag?

    fun findByName(name: String): Tag?

    fun find(): List<Tag>

    fun findAllToStringList(): List<String>

    fun findSelected(): List<SelectedTagDto>

    fun updateSelected(tags: List<String>)

    fun delete(tag: Tag)

    fun find(pageable: Pageable, tagType: Types.TagType?, query: String?): List<TagDto>
}
